### PR TITLE
DM-26771: Allow other calibration objects to be configured

### PR DESCRIPTION
**Pull request by SimonKrughoff**

### CODE REVIEW

**Review by jeffcarlin — APPROVED** [review]
We should confirm that we can run `metrics_pipeline.yaml` on a dataset that has FGCM/jointcal, but otherwise I'm fine with these changes.

File: pipelines/gen_inputs_matched_jointcal_fgcm.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+tasks:
+  matchCatalogs:
+    class: metric_pipeline_tasks.MatchedCatalogTask
+    config:
+      connections.photoCalibName: fgcm_photoCalib
+      apply_external_wcs: True  # We only support jointcal for now
+  matchCatalogsTract:
+    class: metric_pipeline_tasks.MatchedCatalogTractTask
+    config:
+      connections.photoCalibName: fgcm_photoCalib
```
[the remark this answers is not in the document]
Comment (SimonKrughoff) [reply pipelines/gen_inputs_matched_jointcal_fgcm.yaml:11]: I have tried this out on a version of HSC RC2.  It certainly loads the correct datasets.  I did not check the veracity of the application, but that will come out in the verification ticket

File: python/metric_pipeline_tasks/MatchedCatalogsBase.py [hunk 2]
```diff
@@ -44,78 +54,65 @@ class MatchedBaseTask(pipeBase.PipelineTask):
...
-class MatchedTractBaseTask(pipeBase.PipelineTask):
+class MatchedTractBaseTask(MatchedBaseTask):
 
     ConfigClass = MatchedBaseTaskConfig
     _DefaultName = "matchedTractBaseTask"
 
     def __init__(self, config: pipeBase.PipelineTaskConfig, *args, **kwargs):
         super().__init__(*args, config=config, **kwargs)
         self.radius = self.config.match_radius
+        self.level = "tract"
```
[the remark this answers is not in the document]
Comment (SimonKrughoff) [reply python/metric_pipeline_tasks/MatchedCatalogsBase.py:111]: Great!

File: python/metric_pipeline_tasks/MatchedCatalogsBase.py [hunk 3]
```diff
@@ -44,78 +54,65 @@ class MatchedBaseTask(pipeBase.PipelineTask):
     def __init__(self, config: pipeBase.PipelineTaskConfig, *args, **kwargs):
         super().__init__(*args, config=config, **kwargs)
         self.radius = self.config.match_radius
+        self.level = "patch"
 
-    def run(self, source_catalogs, photo_calibs, vIds, wcs, box):
+    def run(self, source_catalogs, photo_calibs, astrom_calibs, vIds, wcs, box, apply_external_wcs):
```
[the remark this answers is not in the document]
Comment (SimonKrughoff) [reply python/metric_pipeline_tasks/MatchedCatalogsBase.py:59]: `astrom_calibs` is always defined, but is sometimes filled with `None`s.  There is checking that if it is set to True and there are no `astrom_calibs` a warning message is emitted.  That seems sufficient

File: pipelines/gen_inputs_matched_jointcal_fgcm.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+description: Produce matched catalogs
```
[the remark this answers is not in the document]
Comment (SimonKrughoff) [reply pipelines/gen_inputs_matched_jointcal_fgcm.yaml:1]: OK.  I'll leave this for another ticket.

**Pull request merged**